jobrunner/lib/database.py

```diff
@@ -168,13 +172,28 @@ def get_connection(filename=None):
     # Looks icky but is documented `threading.local` usage
     cache = CONNECTION_CACHE.__dict__
     if filename not in cache:
-        conn = sqlite3.connect(filename)
+        try:
+            conn = sqlite3.connect(filename)
+        except sqlite3.OperationalError:
+            raise CannotOpenReadonlyDBFile(filename)
```

Comment:
I did consider checking the exception message.  But it might be better to let the exception propagate up and handle it at a higher level, as you allude to below.